spacepy/datamodel.py

```diff
@@ -223,10 +224,373 @@ def meta(self):
         del self.attrs
 
 
-class dmarray(numpy.ndarray, MetaMixin):
+class ISTPArray:
+    """Mixin class for array using ISTP metadata.
+
+    Array types like `dmarray` provide all these methods if attributes
+    of the array use the
+    `ISTP metadata standard <https://spdf.gsfc.nasa.gov/sp_use_of_cdf.html>`_
+
+    Note that some operations that may seem to relate to an array (e.g.
+    uncertainties) may require the use of other arrays in a container;
+    these are in `ISTPContainer`.
+
+    .. versionadded:: 0.5.0
+
+    .. autosummary::
+        ~ISTPArray.plot_as_line
+        ~ISTPArray.replace_invalid
+    .. automethod:: plot_as_line
+    .. automethod:: replace_invalid
+    """
+    attrs: collections.abc.Mapping
+
+    def replace_invalid(self):
+        """Return data from array with invalid values replaced by `~numpy.nan`.
+
+        Makes a copy of the data and, for any values equal to the
+        ``FILLLVAL`` attribute, greater than ``VALIDMAX``, or less than
+        ``VALIDMIN``, replace with NaN.
+
+        Returns
+        -------
+        `~numpy.ndarray`
+            Transformed data
+
+        See Also
+        --------
+        .pycdf.istp.nanfill : an in-place variant
+
+        Notes
+        -----
+        .. versionadded:: 0.5.0
+
+        Comparisons with ``FILLVAL`` are done using `~numpy.isclose` and
+        so may NaN values thare are near, but not identical, to fill.
```

Comment:
I'm having trouble parsing this... is there a typo confusing me?

Reply:
It's using NaN as a verb...changed to "may replace values, hopefully that clarifies?